File: space-game/Scene/SceneService.vb
Imports Data
Imports Entities
Imports TurnTracker

Public Class SceneService
    Implements ISceneService

    Private ReadOnly _entityManager As IEntityManager
    Private ReadOnly _turnTracker As ITurnTracker
    Private ReadOnly _dataProvider As IDataProvider
    Private ReadOnly _constructor As SceneConstructor

    Public Sub New(entityManager As IEntityManager,
                   turnTracker As ITurnTracker,
                   dataProvider As IDataProvider)
        _entityManager = entityManager
        _turnTracker = turnTracker
        _dataProvider = dataProvider
        _constructor = New SceneConstructor(_entityManager, _turnTracker, _dataProvider)
    End Sub

    Public Sub CreateStartingScene() Implements ISceneService.CreateStartingScene
        _constructor.SolSystem()
    End Sub

    Public ReadOnly Property CurrentSceneState As ISceneState Implements ISceneService.CurrentSceneState
        Get
            Dim entities = _entityManager.GetAllEntities()
            Return New SceneState(entities)
        End Get
    End Property

End Class

File: space-game/Scene/SceneService.cs
using Data;
using Entities;
using System.Collections.Generic;
using TurnTracker;

namespace Scene
{
    public class SceneService : ISceneService
    {
        private readonly IEntityManager _entityManager;
        private readonly SceneConstructor _constructor;

        public SceneService(IEntityManager entityManager, ITurnTracker turnTracker, IDataProvider dataProvider)
        {
            _entityManager = entityManager;
            _constructor = new SceneConstructor(_entityManager, turnTracker, dataProvider);
        }

        public void CreateStartingScene()
        {
            _constructor.SolSystem();
        }

        public ISceneState CurrentSceneState
        {
            get
            {
                List<BaseGameEntity> entities = _entityManager.GetAllEntities();
                return new SceneState(entities);
            }
        }
    }
}
